### PR TITLE
Create a new flightrecorder instance per connection

### PR DESCRIPTION
**Pull request by raboof**
#27574

### CODE REVIEW

**Comment by jrudolph** [comment]
On jenkins the formatting failed

File: akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala [hunk 1]
```diff
@@ -312,11 +312,10 @@ private[remote] class ArteryTcpTransport(
     // If something in the inboundConnectionFlow fails, e.g. framing, the connection will be teared down,
     // but other parts of the inbound streams don't have to restarted.
     val newInboundConnectionFlow = {
-      // must create new Flow for each connection because of the FlightRecorder that can't be shared
-      val afr = createFlightRecorderEventSink()
       Flow[ByteString]
         .via(inboundKillSwitch.flow)
-        .via(new TcpFraming(afr))
+        // must create new Flow for each connection because of the FlightRecorder that can't be shared
```
Comment (jrudolph) [inline akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala]: That's not necessary any more, right?

**Review by jrudolph — APPROVED** [review]
LGTM

File: akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala (hunk 1 above)
Comment (raboof) [reply akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala]: oops moved the comment but forgot to update it, fixed

**Review by helena — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**